File: src/Flint.Protocol/BrowserWorkspaceResizeMessages.cs
namespace Flint.Protocol;

/// <summary>Sets column and row dividers in ten-thousandths; requires protocol four.</summary>
public sealed record BrowserWorkspaceResizeMessage(long Epoch, long CommandId, long ExpectedRevision,
    int Column, int Row, byte Mode = 0) : WireMessage
{
    /// <inheritdoc />
    public override int TypeId => (int)WireMessageType.BrowserWorkspaceResize;
}

/// <summary>Authoritative dividers paired with the workspace snapshot revision.</summary>
public sealed record BrowserWorkspaceGeometryMessage(long Epoch, long Revision, int Column, int Row, byte Mode = 0) : WireMessage
{
    /// <inheritdoc />
    public override int TypeId => (int)WireMessageType.BrowserWorkspaceGeometry;
}

File: protocol/kotlin/src/main/kotlin/com/rextechnologies/flint/protocol/wire/BrowserWorkspaceResizeMessages.kt
package com.rextechnologies.flint.protocol.wire

data class BrowserWorkspaceResizeMessage(
    val epoch: Long,
    val commandId: Long,
    val expectedRevision: Long,
    val column: Int,
    val row: Int,
    val mode: Int = 0,
) : WireMessage {
    override val typeId = WireMessageType.BROWSER_WORKSPACE_RESIZE.id
}

data class BrowserWorkspaceGeometryMessage(
    val epoch: Long,
    val revision: Long,
    val column: Int,
    val row: Int,
    val mode: Int = 0,
) : WireMessage {
    override val typeId = WireMessageType.BROWSER_WORKSPACE_GEOMETRY.id
}
